cli/integrity/repository.go

```diff
@@ -0,0 +1,18 @@
+package integrity
+
+import "io"
+
+// Repository provides utilities for working with files and
+// ensuring that they were not compomised.
+type Repository interface {
+	// AddHash extends the list of supported hashes.
+	AddHash(h any)
+	// Add checks signature of provided hashes file, and adds files
+	// to the repository.
+	Add(path string) error
```

Comment:
It looks like this should be specified when constructing a repository, which means it should be in the implementation constructor, and not in the general interface.